src/javascript-wrapper.js

```diff
@@ -56,19 +56,24 @@ var fRunUpdater = function(updater, updaterArgs, vni) {
     // Execute the updater on the VNI context, passing the updater Parameters as the API arguments
     return new Promise(function(resolve) { 
 
+         // console.log('\ncalling updater for ',vni.nodeInstance.componentName,
+         //             ', nodeName=',vni.nodeInstance.nodeName);
          var results = updater.apply(vni, updaterParameters);
          resolve(results);
-         }).then( function( results ) { 
-
-             if (! _.isUndefined(results)) { 
-                 // Got some results back from updater
-                 // If the updater returned anything, set the output state with it
-                 var newState = vni.outputState();
-                 if (newState.lm === oldOutputStateLm) { 
-                     // updater did not modify output state lm - create a new output state with new lm
-                     vni.outputState(createState(vni.vnid, results));
-                 } 
-             }
+
+     }).then(function(results) { 
+         // console.log('updater returned results: ',results);
+
+         if (! _.isUndefined(results)) { 
+             // Got some results back from updater
+             // If the updater returned anything, set the output state with it
+             var newState = vni.outputState();
+             if (newState.lm === oldOutputStateLm || 
+                 _.isUndefined(oldOutputStateLm) && ! _.isUndefined(newState.lm)) { 
```

Comment:
The data field should be set unconditionally here, I believe.  But if the updater also explicitly set the lm already, then I guess we should use that instead of creating a new one:
```
newState.data = results;
if (newState.lm === oldOutputStateLm) { newState.lm = createLm(); }
vni.outputState(newState);  // Not needed, but included for possible future copy-on-write
```